docs/extend/modular-resources/upload/_index.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "Upload your own modules to the Viam Registry"
+linkTitle: "Upload"
+weight: 20
+type: "docs"
+tags: ["server", "rdk", "extending viam", "modular resources", "components", "services"]
+description: "Use the Viam CLI to upload a custom module to the Viam Registry."
+no_list: true
+---
+
+{{% alert title="Beta Notice" color="note" %}}
+This feature is in beta, and may not be suitable for production use.
+{{% /alert %}}
+
+Once you have [created a custom module](/extend/modular-resources/create/), you can use the [Viam CLI](/manage/cli/) to upload it to the Viam Registry.
+
+With the CLI, you can register your module with the Viam Registry to share it with other Viam users, or upload it as a private module that is shared only within your [organization](/manage/fleet/organizations/).
+For more information, see the [`viam module` command](/manage/cli/#module).
+
+## Upload a custom module
+
+To upload your custom module to the Viam Registry, either as a public or private module, follow the instructions below:
+
+1. First, [install the Viam CLI](/manage/cli/#install) and [authenticate](/manage/cli/#authenticate) to Viam.
```

Comment:
1. First, [install the Viam CLI](/manage/cli/#install) and [authenticate](/manage/cli/#authenticate) to Viam. You shoud be installing the CLI on the machine that you intend to use to registry your module code